crates/sui-indexer/src/models_v2/objects.rs

```diff
@@ -161,20 +168,26 @@ impl StoredObject {
                 )))
             }
         };
-        let (name, bcs_name) = if let Some(bcs_name) = self.df_name {
-            let name = bcs::from_bytes(&bcs_name).map_err(|e| {
+        let name = if let Some(field_name) = self.df_name {
```

Comment:
wondering what's the trade-offs between storing the current `field_name` in DB, compared with the `bcs_name` this func returns below?

Reply:
That's a good question, and the answer is that its probably complicated. So today we store a DynamicFieldName which has both the type of the name and a json encode value of the key. This is the encoding that we end up sending on the wire back to clients, but we do also need the bcs form of it. Today to do that we do this complicated dance of looking at the type layout and then taking that and the json to convert back to bcs. I wonder if we could possibly side step that and go straight from json to bcs or figure out something else.

Reply:
If you have any ideas or preferences please let me know!